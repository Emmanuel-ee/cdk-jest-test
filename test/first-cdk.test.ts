import * as cdk from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import * as FirstCdk from '../lib/first-cdk-stack';


// example test. To run these tests, uncomment this file along with the
// example resource in lib/first-cdk-stack.ts
//unit test

test('SQS Queue Created', () => {
  const app = new cdk.App();
  //fine grained
    // WHEN
  const stack = new FirstCdk.FirstCdkStack(app, 'MyTestStack');
    // THEN
  const template = Template.fromStack(stack);

  template.hasResourceProperties('AWS::SQS::Queue', {
    VisibilityTimeout: 300
  });
});

//snapshot test

test('Test for snapshot', () => {
    // Initiate the cdk app
    const app = new cdk.App();
      // create stack
    const stack = new FirstCdk.FirstCdkStack(app, 'MyTestStack');
      // Prepare the stack for assertions
    const template = Template.fromStack(stack);

    //Match with snapshot
    expect(template.toJSON()).toMatchSnapshot()
  });

