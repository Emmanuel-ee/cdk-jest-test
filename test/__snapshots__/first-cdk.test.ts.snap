// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for snapshot 1`] = `
{
  "Outputs": {
    "mybucket": {
      "Value": {
        "Ref": "someBucket8B32029D",
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "duration": {
      "Default": 6,
      "MaxValue": 10,
      "MinValue": 1,
      "Type": "Number",
    },
  },
  "Resources": {
    "FirstCdkQueueC67EAB3D": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "VisibilityTimeout": 300,
      },
      "Type": "AWS::SQS::Queue",
      "UpdateReplacePolicy": "Delete",
    },
    "someBucket8B32029D": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "LifecycleConfiguration": {
          "Rules": [
            {
              "ExpirationInDays": {
                "Ref": "duration",
              },
              "Status": "Enabled",
            },
          ],
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
